syntax = "proto3";
package wx3;
import "ocr_common.proto";

message OcrInputBuffer {
	// 看起来有两种传递图片的方式。
	// 第一种是只传文件路径 pic_path = "C:/path/to/xxx.png"
	// 第二种是用字节流传，猜测需要用到u2,u3,pic_data变量。暂时微信没有用到，可能是为将来保留
	optional string pic_path = 1;
	optional uint32 u2 = 2;
	optional uint32 u3 = 3;
	optional bytes pic_data = 4;
}

message OcrOutputBuffer {
	repeated ocr_common.OCRResultLine lines = 1;	//repeated 每行的结果
	optional uint32 img_width = 2;
	optional uint32 img_height = 3;
	optional string unk4 = 4;
}

message OcrRespond {
	optional int32 type = 1;									// type=1像是初始化成功回调。如果是正常OCR请求，回答的type=0
	optional uint64 task_id = 2;
	optional int32 err_code = 3;
	optional OcrOutputBuffer ocr_result = 4;
}

message OcrRequest {
	int32 type = 1;	 //为0执行ocr，为1会直接返回init信息. 与OcrRespond.type意义相同
	// 经过反复核查，在腾讯proto文件中，这个task_id确实是64位的。但在，在执行过程中，高32位会被丢弃，且第32位为1会出错。
	// 也就是协议上是有64位的uint64，实际上只能有31位。必须是>0的整形数字，范围是[1,2147483647]
	// 由于 task_id = 1会被用于初始化，所以最好取值为 [2, 0x7fffFFFF]
	uint64 task_id = 2;
	OcrInputBuffer input = 3;
}
